### PR TITLE
Add 22.04 skeleton 

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/init-variables.sh [hunk 1]
```diff
@@ -161,6 +161,18 @@ if [[ "$MODE" == "sigMode" || "$MODE" == "gen2Mode" ]]; then
...
+				--resource-group ${AZURE_RESOURCE_GROUP_NAME} \
+				--gallery-name ${SIG_GALLERY_NAME} \
+				--gallery-image-definition ${SIG_IMAGE_NAME} \
+				--publisher microsoft-aks \
+				--offer ${SIG_GALLERY_NAME} \
+				--sku ${SIG_IMAGE_NAME} \
+				--os-type ${OS_TYPE} \
+				--hyper-v-generation ${HYPERV_GENERATION} \
+				--location ${AZURE_LOCATION} \
+				--features SecurityType=TrustedLaunch
```
Comment (chengliangli0918) [inline vhdbuilder/packer/init-variables.sh:175]: Just wondering any reason we need TrustedLaunch for temporary image definition to capture Gen2 22.04 disk? with this option, when the temporary SIG version is converted to VHD blob with SAS in following step, it fails with following error.

Message: The provided gallery image: /subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/2204Gen2/versions/1.1658352289.30585 supports creation of OS disk only with SecurityType as TrustedLaunch.

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 2]
```diff
@@ -469,50 +469,50 @@ stages:
...
       steps:
         - bash: |
             echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
             echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
-            echo '##vso[task.setvariable variable=OS_VERSION]20.04'
+            echo '##vso[task.setvariable variable=OS_VERSION]22.04'
             echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
-            echo '##vso[task.setvariable variable=IMG_OFFER]0001-com-ubuntu-server-focal'
-            echo '##vso[task.setvariable variable=IMG_SKU]20_04-lts-gen2'
+            echo '##vso[task.setvariable variable=IMG_OFFER]0001-com-ubuntu-server-jammy'
```
Comment (alexeldeib) [inline .pipelines/.vsts-vhd-builder-release.yaml:506]: _we're jammin', jammin'..._

File: parts/linux/cloud-init/artifacts/cis.sh [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
 assignRootPW() {
   if grep '^root:[!*]:' /etc/shadow; then
+    VERSION=$(grep DISTRIB_RELEASE /etc/*-release| cut -f 2 -d "=")
     SALT=$(openssl rand -base64 5)
     SECRET=$(openssl rand -base64 37)
-    CMD="import crypt, getpass, pwd; print crypt.crypt('$SECRET', '\$6\$$SALT\$')"
-    HASH=$(python -c "$CMD")
+    CMD="import crypt, getpass, pwd; print(crypt.crypt('$SECRET', '\$6\$$SALT\$'))"
+    if [[ "${VERSION}" == "22.04" ]]; then
+      HASH=$(python3 -c "$CMD")
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cis.sh:10]: suggest to always use python3 and ensure it's installed in vhd, I thought it was already just not symlinked to `python`

File: vhdbuilder/packer/install-dependencies.sh [hunk 4]
```diff
@@ -133,6 +133,11 @@ if [[ ${CONTAINER_RUNTIME:-""} == "containerd" ]]; then
   for version in $containerd_versions; do
     containerd_version="$(echo "$version" | cut -d- -f1)"
     containerd_patch_version="$(echo "$version" | cut -d- -f2)"
+    # containerd 1.4 not available in ubuntu 22.04
+    if [[ ${UBUNTU_RELEASE} == "22.04" && ${containerd_version} == "1.4.13" ]]; then
```
Comment (alexeldeib) [inline vhdbuilder/packer/install-dependencies.sh:137]: let's just stop precaching this for all cases? but this is fine for now

**Review by alexeldeib — APPROVED** [review]
minor comments, mostly lgtm

File: vhdbuilder/packer/init-variables.sh (hunk 1 above)
Comment (anujmaheshwari1) [reply vhdbuilder/packer/init-variables.sh]: its for TL feature support for ACC. we decided to go forward with Vanilla 22.04 and take up TL + 22.04 later.

File: vhdbuilder/packer/install-dependencies.sh (hunk 4 above)
Comment (anujmaheshwari1) [reply vhdbuilder/packer/install-dependencies.sh:137]: that makes sense, ill do it in a separate PR

**Pull request merged**